hasher.go

```diff
@@ -74,44 +86,68 @@ func (vh *valueHasher) ValueHashSize() int {
 	return vh.hasher.Size()
 }
 
-func (th *trieHasher) digest(data []byte) []byte {
+// Path satisfies the PathHasher#Path interface
+func (n *nilPathHasher) Path(key []byte) []byte {
+	return key[:n.hashSize]
+}
+
+// PathSize satisfies the PathHasher#PathSize interface
+func (n *nilPathHasher) PathSize() int {
+	return n.hashSize
+}
+
+// digestData returns the hash of the data provided using the trie hasher.
+func (th *trieHasher) digestData(data []byte) []byte {
 	th.hasher.Write(data)
-	sum := th.hasher.Sum(nil)
+	digest := th.hasher.Sum(nil)
 	th.hasher.Reset()
-	return sum
+	return digest
 }
 
-func (th *trieHasher) digestLeaf(path []byte, leafData []byte) ([]byte, []byte) {
-	value := encodeLeaf(path, leafData)
-	return th.digest(value), value
+// digestLeafNode returns the encoded leaf data as well as its hash (i.e. digest)
+func (th *trieHasher) digestLeafNode(path, data []byte) (digest, value []byte) {
+	value = encodeLeafNode(path, data)
+	digest = th.digestData(value)
+	return
 }
 
-func (th *trieHasher) digestSumLeaf(path []byte, leafData []byte) ([]byte, []byte) {
-	value := encodeLeaf(path, leafData)
-	digest := th.digest(value)
-	digest = append(digest, value[len(value)-sumSize:]...)
-	return digest, value
+func (th *trieHasher) digestInnerNode(leftData, rightData []byte) (digest, value []byte) {
```

Comment:
Either add comments to all of these or none `digestLeafNode` sets the example but the names are self explanatory - your call